sdks/go/pkg/beam/runners/prism/internal/engine/elementmanager.go

```diff
@@ -1246,6 +1305,113 @@ func (ss *stageState) AddPending(newPending []element) int {
 	return len(newPending)
 }
 
+// computeNextTriggeredPane produces the correct pane relative to the previous pane,
+// and the end of window state.
+func computeNextTriggeredPane(pane typex.PaneInfo, endOfWindowReached bool) typex.PaneInfo {
+	// This is the first firing, since index and first are both
+	// set to their zero values.
+	if pane.Index == 0 && !pane.IsFirst {
+		pane.IsFirst = true
+	} else {
+		pane.Index++
```

Comment:
IsFirst is confusing me a bit, but should we be setting `pane.IsFirst = false` in this else condition?

Reply:
Same comment applies below